sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -732,4 +732,20 @@ private[spark] object QueryCompilationErrors {
     new InvalidUDFClassException(s"No handler for UDAF '$name'. " +
       "Use sparkSession.udf.register(...) instead.")
   }
+
+   def unexpectedColumnExpressionError(column: Expression): Throwable = {
+    new AnalysisException(s"[BUG] unexpected column expression: $column")
+  }
+
+  def databaseFromV1SessionCatalogNotSpecifiedError(): Throwable = {
+    new AnalysisException("Database from v1 session catalog is not specified")
+  }
+
+  def nestedDatabaseUnsupportedByV1SessionCatalogError(catalog: String): Throwable = {
+    new AnalysisException(s"Nested databases are not supported by v1 session catalog: $catalog")
+  }
+
+  def invalidCallFunctionOnUnresolvedObjectError(function: String): String = {
```

Comment:
For this one, I think it would be better to update all `new UnresolvedException` instead of adding the error message here. For example
```scala
def invalidFunctionCallOnUnresolvedObjectError(function: String): Throwable = {
  new UnresolvedException(s"Invalid call to $function on unresolved object")
}
```
What do you think?

Reply:
Another PR try to change this, please refer to https://github.com/apache/spark/pull/31337/files#diff-28fdb82e61b32fc0cabfbbace2353cb7849326d3fcfe40567c1ee8a51b53a519R38

Reply:
I think this is a bit strange to have a string in here representing the error message, instead of the error itself. We can either change all `throw new UnresolvedException` to be `throw QueryCompilationError.unresolvedError` or we can leave the UnresolvedException as it is for now.

Reply:
OK